lib/idempotence/reservation/reservation.rb

```diff
@@ -107,17 +107,22 @@ def self.build
       class Reservation
         attr_reader :message
         attr_reader :idempotence_key
-        attr_reader :process_block
 
         def call(message, idempotence_key, &block)
           @message = message
           @idempotence_key = idempotence_key
 
-          yield message if @process_block
+          yield message if process_block
         end
 
-        def set_reserved(value)
-          @process_block = value
+        def process_block
+          @process_block ||= false
+        end
```

Comment:
We can have an attr_reader and initialize `process_block` to false in the initialize method of the substitute class, no need for this memoization.

Reply:
Agreed, this method looks unnecessary. Unset instance variables are naturally falsey

Reply:
agreed